gobblin-data-management/src/main/java/gobblin/data/management/retention/dataset/ConfigurableCleanableDataset.java

```diff
@@ -48,41 +51,40 @@
   private final VersionFinder<? extends T> versionFinder;
   private final RetentionPolicy<T> retentionPolicy;
 
-  public ConfigurableCleanableDataset(FileSystem fs, Properties props, Path datasetRoot) throws IOException {
-    this(fs, props, datasetRoot, LoggerFactory.getLogger(ConfigurableCleanableDataset.class));
-  }
-
+  /**
+   * Creates a new ConfigurableCleanableDataset configured through gobblin-config-management. The constructor expects
+   * {@link #VERSION_FINDER_CLASS_KEY} and {@link #RETENTION_POLICY_CLASS_KEY} to be available in the
+   * <code>config</code> passed.
+   */
   @SuppressWarnings("unchecked")
-  public ConfigurableCleanableDataset(FileSystem fs, Properties props, Path datasetRoot, Logger log)
+  public ConfigurableCleanableDataset(FileSystem fs, Properties jobProps, Path datasetRoot, Config config, Logger log)
       throws IOException {
-    super(fs, props, log);
+
+    super(fs, jobProps, log);
     this.datasetRoot = datasetRoot;
 
-    Preconditions.checkArgument(props.containsKey(VERSION_FINDER_CLASS_KEY),
-        "Missing property " + VERSION_FINDER_CLASS_KEY);
-    Preconditions.checkArgument(props.containsKey(RETENTION_POLICY_CLASS_KEY),
-        "Missing property " + RETENTION_POLICY_CLASS_KEY);
+    Preconditions.checkArgument(config.hasPath(VERSION_FINDER_CLASS_KEY), "Missing property " + VERSION_FINDER_CLASS_KEY);
+    Preconditions.checkArgument(config.hasPath(RETENTION_POLICY_CLASS_KEY), "Missing property " + RETENTION_POLICY_CLASS_KEY);
 
     try {
-      Class<?> versionFinderClass = Class.forName(props.getProperty(VERSION_FINDER_CLASS_KEY));
-      Class<?> retentionPolicyClass = Class.forName(props.getProperty(RETENTION_POLICY_CLASS_KEY));
-      this.versionFinder = (VersionFinder) versionFinderClass.
-          getConstructor(FileSystem.class, Properties.class).newInstance(this.fs, props);
-      this.retentionPolicy = (RetentionPolicy) retentionPolicyClass.
-          getConstructor(Properties.class).newInstance(props);
-    } catch(ClassNotFoundException exception) {
-      throw new IOException(exception);
-    } catch(NoSuchMethodException exception) {
-      throw new IOException(exception);
-    } catch(InstantiationException exception) {
-      throw new IOException(exception);
-    } catch(IllegalAccessException exception) {
-      throw new IOException(exception);
-    } catch(InvocationTargetException exception) {
-      throw new IOException(exception);
+      this.versionFinder =
+          (VersionFinder<? extends T>) ConstructorUtils.invokeConstructor(Class.forName(config.getString(VERSION_FINDER_CLASS_KEY)),
+              new Object[] { this.fs, config });
```

Comment:
You can do `ConstructorUtils.invokeConstructor(Class.forName(config.getString(VERSION_FINDER_CLASS_KEY)), this.fs, config);`